Signal/src/textsecure/Messages/TSMessagesManager.m

```diff
@@ -179,6 +179,21 @@ - (void)handlePreKeyBundle:(IncomingPushMessageSignal*)preKeyMessage{
 }
 
 - (void)handleIncomingMessage:(IncomingPushMessageSignal*)incomingMessage withPushContent:(PushMessageContent*)content{
+    if(content.group!= nil )  {
+        __block BOOL ignoreMessage = NO;
+        [self.dbConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
+            GroupModel *emptyModelToFillOutId = [[GroupModel alloc] initWithTitle:nil memberIds:nil image:nil groupId:content.group.id]; // TODO refactor the TSGroupThread to just take in an ID (as it is all that it uses). Should not take in more than it uses
+            TSGroupThread *gThread = [TSGroupThread threadWithGroupModel:emptyModelToFillOutId transaction:transaction];
+            if(gThread==nil) {
+                ignoreMessage = YES;
+            }
+        }];
+        if(ignoreMessage) {
+            DDLogDebug(@"recevied message from group that I left, ignoring");
```

Comment:
so my proposed change would be the following if wrapper in TSSocketMessage
to listen to what the group said today in my understanding:

- (void)processWebSocketRequestMessage:(WebSocketRequestMessage*)message {

    DDLogInfo(@"Got message with verb: %@ and path: %@", message.verb,
message.path);

    if([message dontIgnore]) {

      [self sendWebSocketMessageAcknowledgement:message];

   }

On Fri, Jan 2, 2015 at 9:05 PM, Christine Corbett Moran <
christine.corbett@gmail.com> wrote:

> if you could read the logs that might be helpful.
>
> On Fri, Jan 2, 2015 at 9:04 PM, Frederic Jacobs <notifications@github.com>
> wrote:
>
>> In Signal/src/textsecure/Messages/TSMessagesManager.m
>> <https://github.com/WhisperSystems/Signal-iOS/pull/248#discussion-diff-22422795>
>> :
>>
>> > @@ -179,6 +179,21 @@ - (void)handlePreKeyBundle:(IncomingPushMessageSignal*)preKeyMessage{
>> >  }
>> >
>> >  - (void)handleIncomingMessage:(IncomingPushMessageSignal*)incomingMessage withPushContent:(PushMessageContent*)content{
>> > +    if(content.group!= nil )  {
>> > +        __block BOOL ignoreMessage = NO;
>> > +        [self.dbConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
>> > +            GroupModel *emptyModelToFillOutId = [[GroupModel alloc] initWithTitle:nil memberIds:nil image:nil groupId:content.group.id]; // TODO refactor the TSGroupThread to just take in an ID (as it is all that it uses). Should not take in more than it uses
>> > +            TSGroupThread *gThread = [TSGroupThread threadWithGroupModel:emptyModelToFillOutId transaction:transaction];
>> > +            if(gThread==nil) {
>> > +                ignoreMessage = YES;
>> > +            }
>> > +        }];
>> > +        if(ignoreMessage) {
>> > +            DDLogDebug(@"recevied message from group that I left, ignoring");
>>
>> Then I'm misunderstanding the issue. :/
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/WhisperSystems/Signal-iOS/pull/248/files#r22422795>.
>>
>
>
>
> --
> Dr. Corbett Moran
> Rockets @ http://www.spacex.com
> Physics @ http://www.ICS.uzh.ch  <http://www.ICS.uzh.ch>
> Dev @ http://circleof6app.com
> Dev @ https://whispersystems.org
> www.christinecorbettmoran.com
>
>


-- 
Dr. Corbett Moran
Rockets @ http://www.spacex.com
Physics @ http://www.ICS.uzh.ch  <http://www.ICS.uzh.ch>
Dev @ http://circleof6app.com
Dev @ https://whispersystems.org
www.christinecorbettmoran.com